executors/cpp/number_fmt.cpp

```diff
@@ -91,35 +305,59 @@ const string test_numfmt(json_object *json_in) {
 
   // Get the numeric value
   double input_double = std::stod(input_string);
-  std:: cout<< "num_double = " << input_double << std::endl;
 
-  // TEMPORARY - just return the input.
-  test_result = input_string;
+  LocalizedNumberFormatter nf;
+  if (notation_string == "scientific") {
+    notation_setting = Notation::scientific();
+  }
 
-  // TODO!!!
-  //  SimpleNumberFormatter snf = SimpleNumberFormatter::forLocale(displayLocale, status);
+  if (style_string == "currency") {
+    // TODO: Generalize
+    nf = NumberFormatter::withLocale(displayLocale)
+         .notation(notation_setting)
+         .unit(CurrencyUnit(currency_string, status))
+         .precision(precision_setting)
+         .integerWidth(integerWidth_setting)
+         .scale(scale_setting)
+         .unit(unit_setting)
+         .unitWidth(unit_width_setting);
+  }
+  else {
+    // Use settings to initialize the formatter
+    nf = NumberFormatter::withLocale(displayLocale)
+         .notation(notation_setting)
+         .precision(precision_setting)
+         .integerWidth(integerWidth_setting)
+         .scale(scale_setting)
+         .unit(unit_setting)
+         .unitWidth(unit_width_setting);
+  }
```

Comment:
To generalize, put the common stuff outside the if statement and then do `nf = nf.unit(...)` inside the if

Reply:
I've revised this to merge the two styles.